lib/ex_unit/lib/ex_unit/cli_formatter.ex

```diff
@@ -210,14 +210,22 @@ defmodule ExUnit.CLIFormatter do
     do: colorize([:bright, :black], msg, config)
 
   defp formatter(:diff_delete, msg, config),
-    do: colorize(:red, msg, config)
+    do: colorize(:red, show_whitespace(msg), config)
 
   defp formatter(:diff_insert, msg, config),
-    do: colorize(:green, msg, config)
+    do: colorize(:green, show_whitespace(msg), config)
 
   defp formatter(_,  msg, _config),
     do: msg
 
+  defp show_whitespace(msg) do
+    if String.match?(msg, ~r/\A[ ]+\z/) do
```

Comment:
It's only one particular case when we have whole fragment as whitespace, it can easily be `[del: " a"]` (e.g. `"a ac"` and `"abbc"`), it means we can't perform substitution here.

> I considered this but thought it would be better to allow different formatters to have different behaviours here.

One option could be it to split string fragments [here](https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/lib/ex_unit/diff.ex#L299) on `" "` and insert it back as `formatter.(:diff_delete, " ")` thus we get strictly whitespace fragments inside `CLIFromatter`. It may also help in handling of other whitespace characters.